pkg/stream/xprotocol/connpool.go

```diff
@@ -19,75 +19,63 @@ package xprotocol
 
 import (
 	"context"
-	"sync"
-	"sync/atomic"
-	"time"
-
-	"mosn.io/mosn/pkg/protocol/xprotocol"
-
 	"mosn.io/api"
 	mosnctx "mosn.io/mosn/pkg/context"
-	"mosn.io/mosn/pkg/log"
 	"mosn.io/mosn/pkg/network"
 	"mosn.io/mosn/pkg/protocol"
-	str "mosn.io/mosn/pkg/stream"
+	"mosn.io/mosn/pkg/protocol/xprotocol"
 	"mosn.io/mosn/pkg/types"
-	"mosn.io/pkg/utils"
+	"sync/atomic"
 )
 
+func init() {
+	network.RegisterNewPoolFactory(protocol.Xprotocol, NewConnPool)
+	types.RegisterConnPoolFactory(protocol.Xprotocol, true)
+}
+
+// for xprotocol
 const (
 	Init = iota
 	Connecting
 	Connected
 )
 
-func init() {
-	network.RegisterNewPoolFactory(protocol.Xprotocol, NewConnPool)
-	types.RegisterConnPoolFactory(protocol.Xprotocol, true)
-}
-
 // types.ConnectionPool
-// activeClient used as connected client
-// host is the upstream
-type connPool struct {
-	activeClients sync.Map //sub protocol -> activeClient
-	host          atomic.Value
-	mux           sync.Mutex
-	tlsHash       *types.HashValue
+// nolint
+type connpool struct {
+	host             atomic.Value
+	tlsHash          *types.HashValue
+	totalClientCount uint64 // total clients
```

Comment:
totalClientCount is not counted in multiplex

Reply:
it's fixed 1